mvp-spec/website-design.md

```diff
@@ -0,0 +1,130 @@
+# Web site integration: design
+
+## Read data
+
+This diagram details the steps needed to read existing cookies from Prebid SSO
+- at server level, **if the website decides to use a specific library**, HTTP redirects can be triggered when needed (referred to as "middleware" in the diagram)
+- at browser level, a Javascript library is used
+  - depending on the context, the JS library call a JSON or "redirect" endpoint on the operator
+  - it relies on the **operator proxy**, a component responsible for building operator URLs to call.
+ It is hosted by the website or by a vendor _on the website's behalf_  (see [landscape](./landscape.md) for context)
+  
+
+### Test support of 3rd party cookies
+
+To test if third party cookies are supported and trigger redirect otherwise, the following logic is used:
+
+1. if an HTTP middleware is used, then
+   1. based on user agent, if the browser is known to **not** support 3PC (ex: Safari) ➡️ consider no 3PC and immediately **HTTP redirect**
+2. in Javascript,
+   1. based on user agent, if the browser is known to **not** support 3PC (ex: Safari) ➡️ consider no 3PC and immediately **javascript redirect**
+3. otherwise, attempt to call JSON endpoint and read existing (3PC) Prebid ID
+   1. at the same time, the operator attempts to write a "test" cookie on .prebidsso.com (ie. attempt to write a 3PC)
+4. if Prebid ID is retrieved, of course it means 3PC **are** supported
+5. if no Prebid ID retrieved, it means either that the user is not known, or that 3PC are not supported
+   1. call operator to attempt to read "test" cookie that was just written
+   2. if success ➡️ 3PC **are** supported, it's just that the user is not known
+   3. if failure ➡️ 3PC are **not** supported, **javascript redirect**
```

Comment:
Is it planned to have this shown in the diagram? But just reviewing the diagram the case where 3PC are not supported in a browser where they should be is clearly not explicit (ex: user blocking 3PC on Chrome).

Reply:
I'm sad to hear it's clearly not explicit 😢
It's the `3d party cookie ok?` check: if it's ok, then everything went as expected. If it's not, it means the 3PC are not supported while we thought they would.
Should I rename it? I don't really have a better wording in mind.